src/adapters/utils/networkUtils.js

```diff
@@ -143,8 +144,8 @@ const processAxiosResponse = (clientResponse) => {
     }
     // (edge case) response and code is not present
     return {
-      response: '',
-      status: 500,
+      response: clientResponse?.response?.data || '',
+      status: getErrorStatusCode(clientResponse, HTTP_STATUS_CODES.INTERNAL_SERVER_ERROR),
```

Comment:
should we have a check on whether the extracted status code falls under valid HTTP_STATUS_CODES.values to make sure contract with rudder-server is strictly maintained ?

Reply:
In case where only `response` is available too

Reply:
>should we have a check on whether the extracted status code falls under valid HTTP_STATUS_CODES.values to make sure contract with rudder-server is strictly maintained ?

Usually when http response is sent, we get valid status-codes only. Any special status-code would usually be present inside the response body(the case with `marketo`). As far as contract with server is concerned, we have destination specific handling taking care of it

Reply:
>In case where only response is available too - L142

The response here is whole `httpResponse` from the http-client(in our case `axios`). It'll be too big an object to send it downstream IMHO

Reply:
Resolving the comment